packages/node/src/lib/construct-payload.ts

```diff
@@ -134,7 +135,8 @@ export function constructPayload(
           {
             pageref: payloadData.routePath
               ? new URL(payloadData.routePath, `${getProto(req)}://${req.headers.host}`).toString()
-              : new URL(req.url, `${getProto(req)}://${req.headers.host}`).toString(),
+              : // req.originalUrl is express specific, req.url is node.js
```

Comment:
Why be inconsistent here depending on the server? I think the express server will also have url, so we could still depend on that.

Reply:
ugh this is why: https://github.com/readmeio/metrics-sdks/blob/31e7f34c5c07fb9a62f2599b0dd9c82fbbdde7dc/packages/node/src/lib/express-middleware.ts#L106-L108. non trivial to fix.

Reply:
Would like to remove the express dependency here and just use node's internal `IncomingMessage` and `ServerResponse` objects. 

Aaron recommended extending this type like so, to add `originalUrl` (which is an express-ism):

```js
interface ExpressRequest {
    originalUrl: string
}

type SupportedIncomingRequestData = ExpressRequest & IncommingMessage
```